### PR TITLE
Changes for the frontend to use a session secret retrieved from AWS S…

### PR DESCRIPTION
**Pull request by MartinH-DI**
Changes for the frontend to use a session secret retrieved from AWS Secrets  Manager

### CODE REVIEW

File: express/src/config/session-storage.ts [hunk 1]
```diff
@@ -10,9 +10,12 @@ function getSessionStorage(): RequestHandler {
     if (sessionStorage.tableName) {
         console.log("Using DynamoDB session storage");
         const DynamoDBStore = connect<SessionData>(session);
+        console.log(`********* Retrieving Session Storage. Secret is "${sessionSecret}". *********`);
```
Comment (CharlesIC) [inline express/src/config/session-storage.ts:13]:
```suggestion
```
We do not want to log a secret....

File: express/src/config/environment.ts [hunk 2]
```diff
@@ -3,6 +3,8 @@ export const googleTagId = process.env.GOOGLE_TAG_ID;
 export const showTestBanner = process.env.SHOW_TEST_BANNER === "true";
 export const region = process.env.AWS_REGION || process.env.AWS_DEFAULT_REGION;
 
+export const sessionSecret = process.env.SESSION_SECRET || "ThisIsNotASecret!OhNoNotAtAll*NotInAnyWay";
```
Comment (CharlesIC) [inline express/src/config/environment.ts:6]:
```suggestion
export const sessionSecret = process.env.SESSION_SECRET;
```
The default should be undefined. In `express/src/config/session-storage.ts`, if dynamo storage is selected but no secret is provided, that should be an error.

File: express/src/config/session-storage.ts (hunk 1 above)
Comment (MartinH-DI) [reply express/src/config/session-storage.ts:13]: Sure; this was a debugging statement that I should have removed.

File: express/src/config/environment.ts (hunk 2 above)
Comment (MartinH-DI) [reply express/src/config/environment.ts:6]: Sure; I'll remove the default.

File: express/src/config/session-storage.ts [hunk 3]
```diff
@@ -11,8 +11,10 @@ function getSessionStorage(): RequestHandler {
         console.log("Using DynamoDB session storage");
         const DynamoDBStore = connect<SessionData>(session);
         return session({
-            secret: "keyboard cat",
+            secret: sessionSecret as string,
```
Comment (CharlesIC) [inline express/src/config/session-storage.ts:14]:
```suggestion
            secret: nonNull(sessionSecret),
```
Type assertions must be avoided as they bypass type checking. We can use the helper method to ensure the value is not null.

**Review by CharlesIC — APPROVED** [review]

**Review by CharlesIC — APPROVED** [review]

**Pull request merged**